crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,409 @@
+    block
+        .items()
+        .into_iter()
+        .enumerate()
+        .map(|(position, item)| NodeWithPosition {
+            position,
+            node: item,
+        })
+        .collect()
+}
+
+// takes the unsorted list and returns the first item that is out of order, and the item it should be placed before
+// returns None if the list is sorted
+fn get_first_out_of_order_pair(
+    items: &[NodeWithPosition],
+) -> Option<(NodeWithPosition, NodeWithPosition)> {
+    if items.is_empty() {
+        return None;
+    }
+
+    let mut first_out_of_order_node: Option<NodeWithPosition> = None;
+
+    // find the first node that's 'less than' the previous node
+    let mut prev_item_info = sort_info(&items[0].node);
+    for item in items.iter() {
+        let item_info = sort_info(&item.node);
+        if item_info.cmp(&prev_item_info) == Ordering::Less {
+            first_out_of_order_node = Some(item.clone());
+            break;
+        }
+        prev_item_info = item_info;
+    }
+
+    // find where that node belongs
+    if let Some(first_out_of_order_node) = first_out_of_order_node {
+        let first_out_of_order_node_info = sort_info(&first_out_of_order_node.node);
+        for item in items.iter() {
+            let item_info = sort_info(&item.node);
+            if item_info.cmp(&first_out_of_order_node_info) == Ordering::Greater {
+                return Some((first_out_of_order_node.clone(), item.clone()));
+            }
+        }
+    }
+
+    None
+}
+
+// returns (vendor prefix, name without vendor prefix) in lowercase for a CssIdentifier
+fn css_identifier_to_prop_name(ident: &CssIdentifier) -> Option<(String, String)> {
+    let tok = ident.value_token().ok()?;
+    let raw_prop_name = tok.token_text_trimmed().text().to_string();
+
+    let prop_lowercase = raw_prop_name.to_lowercase();
+    let prop_prefix = vender_prefix(&prop_lowercase).to_string();
+    let unprefixed_prop = if let Some(unprefixed_slice) = prop_lowercase.strip_prefix(&prop_prefix)
+    {
+        unprefixed_slice.to_string()
+    } else {
+        prop_lowercase
+    };
+
+    Some((prop_prefix, unprefixed_prop))
+}
+
+// returns (vendor prefix, name without vendor prefix) in lowercase for a CssDeclarationWithSemicolon
+fn css_declaration_to_prop_name(
+    decl_with_semicolon: &CssDeclarationWithSemicolon,
+) -> Option<(String, String)> {
+    let prop_name = decl_with_semicolon
+        .declaration()
+        .ok()?
+        .property()
+        .ok()?
+        .as_css_generic_property()?
+        .name()
+        .ok();
+    if let Some(AnyCssDeclarationName::CssIdentifier(ident)) = prop_name {
+        return css_identifier_to_prop_name(&ident);
+    }
+    None
+}
+
+const SORT_INFO_KIND_CUSTOM_PROPERTY: u32 = 1;
+const SORT_INFO_KIND_COMPOSES_PROPERTY: u32 = 2;
+const SORT_INFO_KIND_DECLARATION: u32 = 3;
+const SORT_INFO_KIND_UNKNOWN_DECLARATION: u32 = 4;
+const SORT_INFO_KIND_NESTED_RULE_OR_AT_RULE: u32 = 5;
```

Comment:
Rust `enum` implements sorting out of the box based on their order of appearance, when using the derives `Ord` and `PartialOrd`.

This means that if you have something like this:

```rs
#[derive(Ord, PartialOrd)]
enum Order {
	First,
	Second,
	Third
}
```

`Order::First` will have a lower order than `Order::Third`. You can also go further and represent the enum with integers, *if you need*:

```rs
#[derive(Ord, PartialOrd)]
#[repr(u32)]
enum Order {
	First,
	Second,
	Third
}
```

I don't think you need these constants